cfme/tests/cloud/test_cloud_proxy.py

```diff
@@ -0,0 +1,175 @@
+        the provider proxy was definitely being used and not the default.
+
+        Args:
+            None - Uses only the build in objects
+    """
+    logger.info("Begin Test Override Proxy for Provider {}".format(provider.type))
+
+    @request.addfinalizer
+    def _cleanup_proxy():
+        reset_proxy(appliance, ['default', provider.type], provider)
+
+    logger.info("Test to fail first")
+    settings_local(appliance, ["default"], False)
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    with error.expected('Credential validation was not successful:'):
+        validate_provider(appliance, provider)
+        flash.assert_message_match('Credential validation was successful')
+
+    logger.info("Then test to pass using provider specific override")
+    settings_local(appliance, ['default', provider.type], True)
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    validate_provider(appliance, provider)
+    flash.assert_message_match('Credential validation was successful')
+
+
+def settings_local(appliance, proxy_type, valid=True):
+    """Creates the file string and sends it to the appliance.  If more than one provider,
+        It loops through each item in proxy_type
+
+        Args: proxy_type - is a list that is either or more provider.type and/or 'default'
+              valid - determines if we want the proxy to pass or fail.
+
+    """
+    proxy = cfme_data['proxy_servers']['default']
+    creds = credentials[proxy['credentials']]
+    yml_header = ":http_proxy: \n  "
+    for pt in proxy_type:
+        if len(proxy_type) > 1 and pt == 'default':
+            port = '9999'
+        else:
+            port = proxy['port'] if valid else '8686'
+        yml_provider = ":{pt}: \n    :host: {host} \n    ".format(pt=pt, host=proxy['host'])
+        yml_settings = ":password: {pw} \n    :port: {port} \n    :user: {user} ".format(
+            pw=creds['password'], port=port, user=creds['username'])
+        config_yml = yml_header + yml_provider + yml_settings
+    appliance.ssh_client.run_command(
+        "echo \'{config}\' > /var/www/miq/vmdb/config/settings.local.yml".format(config=config_yml))
+
+
+def reset_proxy(appliance, proxy_type, provider):
+    """Clears the file string and sends it to the appliance
+
+        Args: proxy_type - is a list that is either or more provider.type and/or 'default'
+              We have to explicity clear each change as the values are cached at restart.
+    """
+    local_yml = ":http_proxy: \n  "
+    for pt in proxy_type:
+        local_yml += ":{}: \n    :host: \n    :password: \n    :port: \n    :user: \n  ".format(pt)
+    appliance.ssh_client.run_command(
+        "echo \'{}\' > /var/www/miq/vmdb/config/settings.local.yml".format(local_yml))
+    appliance.restart_evm_service(wait_for_web_ui=True)
+    validate_provider(appliance, provider)
+
+
+def validate_provider(appliance, provider):
```

Comment:
Those are complete overkill.  I just need to force a connection attempt to the provider and this gives immediately feedback on whether the port was available or not.  I'm open to any other ways of doing this or adding a new provider object method.  I can see a lot of upcoming tests which would want functionality like this if it doesn't already exist.  Is there anything else like that that doesn't return false positives?